src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -376,8 +375,17 @@ private static string[] ParseArgv(ProcessStartInfo psi, string alternativePath =
                     argvList.Add("openURL"); // kfmclient needs OpenURL
                 }
             }
+
             argvList.Add(argv0);
-            ParseArgumentsIntoList(psi.Arguments, argvList);
+            if (!string.IsNullOrEmpty(psi.Arguments))
+            {
+                ParseArgumentsIntoList(psi.Arguments, argvList);
+            }
+            else
+            {
+                argvList.AddRange(psi.ArgumentList);
+                psi.Arguments = PasteArguments.Paste(psi.ArgumentList, true);
```

Comment:
You need to pass `false` here. `psi.Arguments` does not include the `argv[0]` parameter. In fact, you may want to import just `PasteArguments.cs` and call `AppendArgument` in a loop. The special handling for `argv[0]` and the Unix version of the `Paste` routine is dead code as far this assembly is concerned.